src/inversion_scripts/operators/operator_utilities.py

```diff
@@ -52,15 +80,95 @@ def read_geoschem(date, gc_cache, build_jacobian=False, sensi_cache=None):
-        sensitivities = sensitivity_data["Sensitivities"].values
+        
+        elements = range(n_elements)
+        ntracers = config['NumJacobianTracers']
+        opt_OH = config['OptimizeOH']
+        opt_BC = config['OptimizeBCs']
+
+        n_base_runs = (n_elements - int(opt_OH) - (int(opt_BC) * 4)) / ntracers
+        
+        nruns = (
+            np.ceil(n_base_runs).astype(int) +
+            int(opt_OH) +
+            (int(opt_BC) * 4)
+        )
+        
+        # Dictionary that stores mapping of state vector elements to
+        # perturbation simulation numbers
+        pert_simulations_dict = {}
+        for e in elements:
+            # State vector elements are numbered 1..nelements
+            sv_elem = e + 1
+        
+            is_OH_element = check_is_OH_element(sv_elem, n_elements, opt_OH)
+            # Determine which run directory to look in
+            if is_OH_element:
+                run_number = nruns
+            elif check_is_BC_element(sv_elem, n_elements, opt_OH, opt_BC, is_OH_element):
+                num_back = n_elements % sv_elem
+                run_number = nruns - num_back
+            else:
+                run_number = np.ceil(sv_elem / ntracers).astype(int)
+        
+            run_num = str(run_number).zfill(4)
+        
+            # add the element to the dictionary for the relevant simulation number
+            if run_num not in pert_simulations_dict:
+                pert_simulations_dict[run_num] = [sv_elem]
+            else:
+                pert_simulations_dict[run_num].append(sv_elem)
+                
+                
+        gc_date = pd.to_datetime(date, format='%Y%m%d_%H')
+        ds_all = [concat_tracers(k, gc_date, config, v) for k,v in pert_simulations_dict.items()]
+        ds_sensi = xr.concat(ds_all, 'element')
+        ds_sensi.load()
+        
+        sensitivities = ds_sensi["ch4"].values
         # Reshape so the data have dimensions (lon, lat, lev, grid_element)
         sensitivities = np.einsum("klji->ijlk", sensitivities)
-        sensitivity_data.close()
-        dat["Sensitivities"] = sensitivities
+        dat["jacobian_ch4"] = sensitivities
+        
+        # get emis base, which is also BC base
+        ds_emis_base = concat_tracers('0001', gc_date, config, [0], baserun=True)
+        ds_emis_base.load()
+        dat['emis_base_ch4'] = np.einsum('klji->ijlk', ds_emis_base['ch4'].values)
+        
+        # get OH base, run RunName_0000
+        # it's always here whether OptimizeOH is true or not
+        # so we can keep it here for convenience
+        ds_oh_base = concat_tracers('0000', gc_date, config, [0])
+        ds_oh_base.load()
+        dat['oh_base_ch4'] = np.einsum('klji->ijlk', ds_oh_base['ch4'].values)
+        
 
     return dat
 
+
+def concat_tracers(run_id, gc_date, config, sv_elems, baserun=False):
+    prefix = config['OutputPath'] + '/' + config['RunName'] + '/jacobian_runs'
```

Comment:
can you add a high level description here with a docstring

Reply:
Added a docstring: https://github.com/geoschem/integrated_methane_inversion/pull/252/commits/07e05a19bd0e48c36442d712feb2567a033b3162